LiteCore/Database/PrebuiltCopier.cc

```diff
@@ -0,0 +1,84 @@
+//
+//  PrebuiltCopier.cc
+//  LiteCore
+//
+//  Created by Jim Borden on 2017/08/15.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#include "PrebuiltCopier.hh"
+#include "StringUtil.hh"
+#include "Error.hh"
+#include <thread>
+
+namespace litecore {
+    bool CopyPrebuiltDB(const litecore::FilePath &from, const litecore::FilePath &to,
+                             const C4DatabaseConfig *config, C4Error* outError) {
+        if(!from.exists()) {
+            Warn("No database exists at %s, cannot copy!", from.path().data());
+            outError->code = ENOENT;
+            outError->domain = POSIXDomain;
+            return false;
+        }
+        
+        FilePath backupPath;
+        Log("Copying prebuilt database from %s to %s", from.path().data(), to.path().data());
+        bool needBackup = to.exists();
+
+        try {
+            FilePath temp = FilePath::tempDirectory().mkTempDir();
+            temp.delRecursive();
+            from.copyTo(temp);
+            
+            auto db = c4db_open(c4str(temp.path().data()), config, outError);
```

Comment:
`data()` just points to the characters; it doesn't say anything about what comes after the last character. `c_str()` explicitly promises there is a 0 byte after the last character.

In practice, they return the same pointer, because the string object's internal buffer is nul-terminated. I think this has been made explicit in C++14 or C++17. But even so I think it's best to call `c_str()` if you require a nul-terminated string.